aten/src/ATen/native/cuda/SpectralOps.cu

```diff
@@ -418,10 +438,27 @@ Tensor _fft_cufft(const Tensor& self, int64_t signal_ndim,
   // (see kRoundSmall and kRoundLarge in THCCachingAllocator.cpp), but we do
   // need to check input tensor to make sure that it is not unaligned, e.g.,
   // from a slicing.
+  bool must_clone = false;
   auto complex_size_bytes = 2 * input.element_size();
   if (reinterpret_cast<std::uintptr_t>(input.data_ptr()) % complex_size_bytes != 0) {
-    input = input.clone(at::MemoryFormat::Contiguous);
-    input_was_cloned = true;
+    must_clone = true;
+  }
+
+  if (complex_input) {
+    auto strides = input.strides();
+    // Real/imag dimension must be like complex type.
+    must_clone |= strides.back() != 1;
+    // Strides of other dimensions needs to be aligned when viewed as complex
+    // type, i.e., multiples of 2.
+    must_clone |= std::any_of(strides.begin(), strides.end() - 1,
+                              [&](int64_t stride) { return stride % 2 != 0; });
+
+    // Complex to real FFTs may overwrite the input buffer (gh-34551)
+    must_clone |= !complex_output;
+  }
```

Comment:
As well as making `CuFFTConfig` tensor-independent, I'm also removing assumptions from the `_fft_with_size` format e.g. here the real tensor representation of complex values. The config strides are now based on complex dtypes and it no longer assumes the output values will be contiguous.